IHP/View/CSSFramework.hs

```diff
@@ -93,52 +93,104 @@ instance Default CSSFramework where
-                    renderCheckboxFormField formField@(FormField {fieldType, fieldName, fieldLabel, fieldValue, validatorResult, fieldClass, disabled, disableLabel, disableValidationResult, fieldInput, labelClass, required, autofocus }) = do
-                        formGroup do
-                            (H.div ! A.class_ "form-check") do
-                                let element = if disableLabel then H.div else H.label ! A.class_ (if labelClass == "" then "form-check-label" else H.textValue labelClass)
-                                element do
-                                    let theInput = H.input
-                                            ! A.type_ "checkbox"
-                                            ! A.name fieldName
-                                            ! A.class_ (cs $ classes ["form-check-input", (inputInvalidClass, isJust validatorResult), (fieldClass, not (null fieldClass))])
-                                            !? (required, A.required "required")
-                                            !? (autofocus, A.autofocus "autofocus")
-                                            !? (fieldValue == "yes", A.checked "checked")
-                                            !? (disabled, A.disabled "disabled")
-                                    theInput
-                                    H.input ! A.type_ "hidden" ! A.name fieldName ! A.value (cs $ inputValue False)
-                                    Blaze.text fieldLabel
-                                    validationResult
-                                    helpText
-
-                    renderTextField :: Blaze.AttributeValue -> FormField -> Blaze.Html
+                    renderCheckboxFormField formField@(FormField {fieldType, fieldName, fieldLabel, fieldValue, fieldInputId, validatorResult, fieldClass, disabled, disableLabel, disableValidationResult, fieldInput, labelClass, required, autofocus }) = do
+                        formGroup [hsx|<div class="form-check">{element}</div>|]
+                        where
+                            theInput = [hsx|
+                                            <input
+                                                type="checkbox"
+                                                name={fieldName}
+                                                class={classes ["form-check-input", (inputInvalidClass, isJust validatorResult), (fieldClass, not (null fieldClass))]}
+                                                id={fieldInputId}
+                                                checked={fieldValue == "yes"}
+                                                required={required}
+                                                disabled={disabled}
+                                                autofocus={autofocus}
+                                            />
+
+                                            <input type="hidden" name={fieldName} value={inputValue False} />
```

Comment:
Why do we need the hidden input?

Reply:
If the checkbox is checked off, the browser will not send the parameter as part of the form. This will then make it impossible to set a field to `False` using a checkbox.

Now when the checkbox is ignored because it's checked off, the hidden field with `value={inputValue False}` is used to set the parameter to `False`